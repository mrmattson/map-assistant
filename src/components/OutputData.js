import React from 'react'
import PropTypes from 'prop-types'
import turfHelpers from '@turf/helpers'
import geojson2kml from '../ShapeTools/Geojson2Kml'

function OutputData ({ shapes }) {
  const kmlString = geojson2kml(turfHelpers.featureCollection(shapes))

  return (
    <div className="OutputData">
      <h4>KML Output</h4>
      <textarea className="OutputData-textarea" value={kmlString} readOnly />
    </div>
  )
}

// https://facebook.github.io/react/docs/typechecking-with-proptypes.html
OutputData.propTypes = {
//   prop: React.PropTypes.string,
//   eslint-disable-next-line react/forbid-prop-types
  shapes: PropTypes.array.isRequired
}

export default OutputData
