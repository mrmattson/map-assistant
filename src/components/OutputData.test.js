/* eslint-env jest */

import React from 'react'
import ReactDOM from 'react-dom'
import renderer from 'react-test-renderer'

import OutputData from './OutputData'

it('renders without crashing', () => {
  const div = document.createElement('div')
  ReactDOM.render(
    <OutputData
      shapes={[]}
    />, div)
})

it('clean snapshot', () => {
  const component = renderer.create(
    <OutputData
      shapes={[]}
    />
  )
  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})
