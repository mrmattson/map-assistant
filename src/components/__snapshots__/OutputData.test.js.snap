exports[`test clean snapshot 1`] = `
<div
  className="OutputData">
  <h4>
    KML Output
  </h4>
  <textarea
    className="OutputData-textarea"
    readOnly={true}
    value="<?xml version=\"1.0\" encoding=\"UTF-8\"?><kml xmlns=\"http://www.opengis.net/kml/2.2\"><Document><name>Generated from GeoConverter</name><description>This data is generated from data input into Geo File Converter.</description></Document></kml>" />
</div>
`;
